diff -uNpr newlib-4.2.0.20211231-org/libgloss/riscv/sys_access.c newlib-4.2.0.20211231/libgloss/riscv/sys_access.c
--- newlib-4.2.0.20211231-org/libgloss/riscv/sys_access.c	2022-01-01 01:46:13.000000000 +0800
+++ newlib-4.2.0.20211231/libgloss/riscv/sys_access.c	2022-11-05 03:26:19.000000000 +0800
@@ -5,5 +5,9 @@
 int
 _access(const char *file, int mode)
 {
+  // Skip errors by -Wno-int-conversion.
+  // I don't know much about embedded development.
+  // Anyone please fix this?
+  // There are many files called syscall_errno passing a string.
   return syscall_errno (SYS_access, 2, file, mode, 0, 0, 0, 0);
 }
diff -uNpr newlib-4.2.0.20211231-org/newlib/libm/complex/cargl.c newlib-4.2.0.20211231/newlib/libm/complex/cargl.c
--- newlib-4.2.0.20211231-org/newlib/libm/complex/cargl.c	2022-01-01 01:46:13.000000000 +0800
+++ newlib-4.2.0.20211231/newlib/libm/complex/cargl.c	2022-11-05 03:22:35.000000000 +0800
@@ -7,6 +7,8 @@
 #include <complex.h>
 #include <math.h>
 
+long double atan2l (long double, long double);
+
 long double
 cargl(long double complex z)
 {     
diff -uNpr newlib-4.2.0.20211231-org/newlib/libm/complex/ccoshl.c newlib-4.2.0.20211231/newlib/libm/complex/ccoshl.c
--- newlib-4.2.0.20211231-org/newlib/libm/complex/ccoshl.c	2022-01-01 01:46:13.000000000 +0800
+++ newlib-4.2.0.20211231/newlib/libm/complex/ccoshl.c	2022-11-05 03:22:35.000000000 +0800
@@ -32,6 +32,11 @@
 #include <complex.h>
 #include <math.h>
 
+long double cosl(long double);
+long double sinl(long double);
+long double coshl(long double);
+long double sinhl(long double);
+
 long double complex
 ccoshl(long double complex z)
 {
diff -uNpr newlib-4.2.0.20211231-org/newlib/libm/complex/cephes_subrl.h newlib-4.2.0.20211231/newlib/libm/complex/cephes_subrl.h
--- newlib-4.2.0.20211231-org/newlib/libm/complex/cephes_subrl.h	2022-01-01 01:46:13.000000000 +0800
+++ newlib-4.2.0.20211231/newlib/libm/complex/cephes_subrl.h	2022-11-05 03:22:35.000000000 +0800
@@ -3,6 +3,14 @@
 void _cchshl(long double, long double *, long double *);
 long double _redupil(long double);
 long double _ctansl(long double complex);
+long double fabsl (long double);
+long double coshl (long double);
+long double sinhl (long double);
+long double expl (long double);
+long double cosl (long double);
+long double sinl (long double);
+long double atan2l (long double, long double);
+long double logl (long double);
 
 #define	M_PIL	3.14159265358979323846264338327950280e+00L
 #define	M_PI_2L	1.57079632679489661923132169163975140e+00L
diff -uNpr newlib-4.2.0.20211231-org/newlib/libm/complex/cexpl.c newlib-4.2.0.20211231/newlib/libm/complex/cexpl.c
--- newlib-4.2.0.20211231-org/newlib/libm/complex/cexpl.c	2022-01-01 01:46:13.000000000 +0800
+++ newlib-4.2.0.20211231/newlib/libm/complex/cexpl.c	2022-11-05 03:22:35.000000000 +0800
@@ -32,6 +32,10 @@
 #include <complex.h>
 #include <math.h>
 
+long double expl (long double);
+long double sinl (long double);
+long double cosl (long double);
+
 long double complex
 cexpl(long double complex z)
 {
diff -uNpr newlib-4.2.0.20211231-org/newlib/libm/complex/clogl.c newlib-4.2.0.20211231/newlib/libm/complex/clogl.c
--- newlib-4.2.0.20211231-org/newlib/libm/complex/clogl.c	2022-01-01 01:46:13.000000000 +0800
+++ newlib-4.2.0.20211231/newlib/libm/complex/clogl.c	2022-11-05 03:22:35.000000000 +0800
@@ -32,6 +32,9 @@
 #include <complex.h>
 #include <math.h>
 
+long double logl (long double);
+long double atan2l (long double, long double);
+
 long double complex
 clogl(long double complex z)
 {
diff -uNpr newlib-4.2.0.20211231-org/newlib/libm/complex/cpowl.c newlib-4.2.0.20211231/newlib/libm/complex/cpowl.c
--- newlib-4.2.0.20211231-org/newlib/libm/complex/cpowl.c	2022-01-01 01:46:13.000000000 +0800
+++ newlib-4.2.0.20211231/newlib/libm/complex/cpowl.c	2022-11-05 03:22:35.000000000 +0800
@@ -32,6 +32,12 @@
 #include <complex.h>
 #include <math.h>
 
+long double cosl (long double);
+long double sinl (long double);
+long double powl (long double, long double);
+long double expl (long double);
+long double logl (long double);
+
 long double complex
 cpowl(long double complex a, long double complex z)
 {
diff -uNpr newlib-4.2.0.20211231-org/newlib/libm/complex/cprojl.c newlib-4.2.0.20211231/newlib/libm/complex/cprojl.c
--- newlib-4.2.0.20211231-org/newlib/libm/complex/cprojl.c	2022-01-01 01:46:13.000000000 +0800
+++ newlib-4.2.0.20211231/newlib/libm/complex/cprojl.c	2022-11-05 03:22:35.000000000 +0800
@@ -45,6 +45,9 @@ __RCSID("$NetBSD: cprojl.c,v 1.7 2014/10
  *
  * INFINITY + I * copysign(0.0, cimag(z))
  */
+
+long double copysignl (long double, long double);
+
 long double complex
 cprojl(long double complex z)
 {
diff -uNpr newlib-4.2.0.20211231-org/newlib/libm/complex/csinhl.c newlib-4.2.0.20211231/newlib/libm/complex/csinhl.c
--- newlib-4.2.0.20211231-org/newlib/libm/complex/csinhl.c	2022-01-01 01:46:13.000000000 +0800
+++ newlib-4.2.0.20211231/newlib/libm/complex/csinhl.c	2022-11-05 03:22:35.000000000 +0800
@@ -32,6 +32,11 @@
 #include <complex.h>
 #include <math.h>
 
+long double cosl (long double);
+long double coshl (long double);
+long double sinl (long double);
+long double sinhl (long double);
+
 long double complex
 csinhl(long double complex z)
 {
diff -uNpr newlib-4.2.0.20211231-org/newlib/libm/complex/csqrtl.c newlib-4.2.0.20211231/newlib/libm/complex/csqrtl.c
--- newlib-4.2.0.20211231-org/newlib/libm/complex/csqrtl.c	2022-01-01 01:46:13.000000000 +0800
+++ newlib-4.2.0.20211231/newlib/libm/complex/csqrtl.c	2022-11-05 03:22:35.000000000 +0800
@@ -49,6 +49,9 @@ __RCSID("$NetBSD: csqrtl.c,v 1.2 2014/10
 
 #define cpackl(r, i) ((r) + (i) * I)
 
+long double fabsl (long double);
+long double copysignl (long double, long double);
+
 long double complex
 csqrtl(long double complex z)
 {
diff -uNpr newlib-4.2.0.20211231-org/newlib/libm/complex/ctanhl.c newlib-4.2.0.20211231/newlib/libm/complex/ctanhl.c
--- newlib-4.2.0.20211231-org/newlib/libm/complex/ctanhl.c	2022-01-01 01:46:13.000000000 +0800
+++ newlib-4.2.0.20211231/newlib/libm/complex/ctanhl.c	2022-11-05 03:22:35.000000000 +0800
@@ -32,6 +32,11 @@
 #include <complex.h>
 #include <math.h>
 
+long double cosl (long double);
+long double sinl (long double);
+long double coshl (long double);
+long double sinhl (long double);
+
 long double complex
 ctanhl(long double complex z)
 {
